app/components/explore/side-bar.js

```diff
@@ -1,13 +1,15 @@
 import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { getDateRanges } from 'open-event-frontend/utils/dictionary/filters';
+import { observer } from '@ember/object';
+import moment from 'moment';
 
 export default Component.extend({
 
   classNames: ['ui', 'fluid', 'explore', 'vertical', 'menu'],
 
-  dateRanges: computed(function() {
-    return getDateRanges.bind(this)();
+  dateChange: observer('start_date', 'end_date', function() {
```

Comment:
Do not use an observer for this. Always use an observer only when there's really no other option.

In this case, you should be using a one way binding for the date picker and make it such that it can accept a call back action with the selected date value. (similar to how the ui-dropdown component works) In that action you can do whatever you do here.

This way the data also flows correctly and is easy to catch bugs.

Observers can easily cause weird issues  that there's even an eslint rule to prevent its usage :P https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-observers.md

Reply:
@niranjan94 But I will still need an observer in the date-picker, right? I think I am not getting the required implementation here. I need something like this in the date-picker right -
```javascript
[observe for change in start_date] {
  this.sendAction('callback_name');
}
```